### PR TITLE
Ethan region explanation

### PR DESCRIPTION
**Pull request by EthanJonesUNSW**
A new documentation page covering Region in detail.

### CODE REVIEW

**Comment by jayenashar** [comment]
i'm still opposed to separating the code from the code documentation

why not use markdown in the code repo?

https://github.com/UNSWComputing/rUNSWift/pull/2406#discussion_r545411963

**Comment by EthanJonesUNSW** [comment]
Again, I still don't think these PRs are the place for that discussion. Right now this is how we have things set out. If we want to change it we'll have to discuss it as a team. If we do decide to change that would be another PR (or several).

**Comment by jayenashar** [comment]
> Again, 

You're previous github comment started talking about code comments and I thought you didn't understand my suggestion to not use code comments.  https://github.com/UNSWComputing/rUNSWift/pull/2406#discussion_r542140909  And we haven't had a meeting in this time, but it's on the agenda for the next meeting.

> I still don't think these PRs are the place for that discussion. 

PRs may not be the best place for the discussion, but it's where the discussion is triggered, and that's how it ended up here (or in https://github.com/UNSWComputing/rUNSWift/pull/2406#discussion_r542112068 ).

PRs are more permanent, so it's a better place to have the record of why this is here, or why it's done this way or that.  E.g., https://github.com/UNSWComputing/rUNSWift/pull/2401#issuecomment-723400797

> Right now this is how we have things set out.

Actually it's not.  The current docs repo is not tightly knit with the code.  This PR is.  We do not have it set out to tightly bind the docs with several references to the code.

> If we want to change it we'll have to discuss it as a team.

We rarely discuss things as a team.  It's usually just the one person who is willing to put in the changes needed that decides and implements it.  Even when we do discuss things as a team, the PR overrides it.  E.g., https://github.com/UNSWComputing/rUNSWift/pull/2401#issuecomment-723400797

> If we do decide to change that would be another PR (or several).

Not really, I'm only suggesting code documentation stays with the code.  The current docs about the runswift codebase are quite high-level.

**Comment by EthanJonesUNSW** [comment]
I've been too indirect I think. Text tends to sound hostile even when it isn't, so part of it is trying to move this somewhere where I can talk with less risk of that. So, here's the more blunt format:

I don't care how we do docs. I don't care if what I've written remains in read the docs or in a readme, both ways are fine. I don't even remember why we switched from the wiki to read the docs as I didn't care back then either.

The ethan-region-cleanup branch has nothing to do with docs. I only wrote new comments for where I added new functions, plus some small tweaks. I just put in a place for a link to help out anyone who opens the file and gets confused. I don't care if that link is changed, the full docs are put somewhere relevant nearby or it just says "For details look at the readme." 

I wrote this new docs section because I know not enough people on the team understand RegionI, so we need comprehensive documentation somewhere. I don't care where, as long as it can be found by new team members even if there is a significant loss of continuity between teams over a couple of years.

In summary, I don't care about the format of the docs and neither of these branches have much to do with it. I genuinely don't like using PRs for tangential discussions. I'd prefer to keep PRs to topics related to whether or not a branch should be merged or changes that need to be made before it is merged. Could we open an empty PR and put the discussion there or something?

**Comment by pzrq** [comment]
@Astilious These are great docs, thanks for putting them together - I’m already learning so reckon they’ll be great to be able to empower future users of the codebase!

It sounds like there’s a lot of common ground here already.

> I don't care how we do docs. I don't care if what I've written remains in read the docs or in a readme, both ways are fine.

One example of where we use the in-folder README pattern (I’m on a phone at the moment and I think it still works well...I’m sure there are plenty more though this was the first one I found): https://github.com/UNSWComputing/rUNSWift-2019-release/tree/master/image/home/nao/data

---

Thus I’d suggest this PR is not needed and can be closed, or perhaps summarised into a few key overview paragraphs with these details in said README (GitHub allows several different formats, so possibly even rst? I’d have to test it...but yeah on a phone...)

Hope that helps. Please feel welcome to let me know if there are any further concerns here, thanks!

**Comment by jayenashar** [comment]
sorry for the late reply.  crazy few days.

sorry wasn't trying to make extra work.  was just making sure my voice was heard as i don't believe it was understood in the other PR.

since github does support rst, we could move the new file from this PR to the main code base without much extra effort, if that is what was decided in the meeting.

**Comment by pzrq** [comment]
✅ Closing as duplicate, thanks @Astilious @jayenashar 

I'm not going to spend the time to figure out exactly how right now, but this appears to be a duplicate (at least all the same headings) as https://github.com/UNSWComputing/rUNSWift-2023-release/blob/main/robot/perception/vision/Region/README.md
